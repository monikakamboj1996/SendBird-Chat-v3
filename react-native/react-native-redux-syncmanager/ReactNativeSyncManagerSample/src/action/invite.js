
const Action = {
	Type: {
		USER_FETCH: 'user-fetch'
	}
};
Action.fetchUsers = users => {
	return {
		type: Action.Type.USER_FETCH,
		payload: { users }
	};
};

export default Action;