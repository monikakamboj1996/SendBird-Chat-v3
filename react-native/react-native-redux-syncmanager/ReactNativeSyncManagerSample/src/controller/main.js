import React from 'react';
import { Alert, Platform, View } from 'react-native';
import { Button } from 'react-native-elements';
import { connect } from 'react-redux';

import AsyncStorage from '@react-native-community/async-storage';
import { FlatList } from 'react-native-gesture-handler';
import ActionButton from 'react-native-action-button';
import Icon from 'react-native-vector-icons/Ionicons';
import firebase from 'react-native-firebase';

import SendBird from 'sendbird';
import SendBirdSyncManager from 'sendbird-syncmanager';

import { style } from '../style/main';
import Action from '../action/main';
import GroupChannelView from '../view/groupChannel';
import { navigator } from '../navigator';

export class MainController extends React.Component {
  static navigationOptions = ({ navigation }) => {
    return {
      title: 'SendBird',
      headerRight: (
        <Button
          containerStyle={style.signoutButtonContainer}
          buttonStyle={style.signoutButton}
          icon={<Icon name="md-log-out" color="#7d62d9" size={28}></Icon>}
          onPress={() => {
            Alert.alert('Sign out', 'Do you want to sign out?', [
              { text: 'Cancel' },
              {
                text: 'OK',
                onPress: () => {
                  AsyncStorage.removeItem('currentUser', err => {
                    SendBirdSyncManager.getInstance().reset();
                    const sb = SendBird.getInstance();
                    firebase
                      .messaging()
                      .getToken()
                      .then(fcmToken => {
                        if (fcmToken) {
                          if (Platform.OS === 'ios') {
                            sb.unregisterAPNSPushTokenForCurrentUser(fcmToken, (result, err) => {
                              sb.disconnect(() => {
                                navigation.replace('Signin', {});
                              });
                            });
                          } else {
                            sb.unregisterGCMPushTokenForCurrentUser(fcmToken, (result, err) => {
                              sb.disconnect(() => {
                                navigation.replace('Signin', {});
                              });
                            });
                          }
                        } else {
                          sb.disconnect(() => {
                            navigation.replace('Signin', {});
                          });
                        }
                      });
                  });
                }
              }
            ]);
          }}
        />
      )
    };
  };
  static getDerivedStateFromProps(props, state) {
    state.channels = props.channels;
    return state;
  }
  constructor(props) {
    super(props);
    this.collection = null;
    this.state = {
      channels: []
    };
  }
  componentDidMount() {
    const { navigation } = this.props;
    const userFromParams = navigation.getParam('user', null);
    if (!userFromParams) {
      AsyncStorage.getItem('currentUser', (err, result) => {
        if (result) {
          const user = JSON.parse(result);
          this.init(user.userId);
        } else {
          this.requireLogin();
        }
      });
    } else {
      this.init(userFromParams.userId);
    }
  }
  componentWillUnmount() {
    if (this.onTokenRefreshListener) {
      this.onTokenRefreshListener();
    }
    if (this.onPushNotificationListener) {
      this.onPushNotificationListener();
    }
    if (this.onPushNotificationOpened) {
      this.onPushNotificationOpened();
    }
    this.props.clearChannels();
    const manager = SendBirdSyncManager.getInstance();
    if (manager) {
      manager.clearCache();
    }
    if (this.collection) {
      this.collection = null;
    }
  }
  init(userId) {
    const sb = SendBird.getInstance();
    sb.connect(userId, (_, err) => {
      if (!err) {
        firebase
          .messaging()
          .getToken()
          .then(fcmToken => {
            if (fcmToken) {
              this.registerPushToken(fcmToken);
            }
          });
      }
    });

    firebase
      .messaging()
      .hasPermission()
      .then(enabled => {
        if (!enabled) {
          firebase
            .messaging()
            .requestPermission()
            .then(() => {
              this.addPushNotificationListener();
            })
            .catch(err => {
              console.log('Failed to get permission', err);
            });
        } else {
          this.addPushNotificationListener();
        }
      });
    this.addPushTokenRefreshListener();

    this.onPushNotificationOpened = firebase.notifications().onNotificationOpened(notificationOpen => {
      firebase.notifications().removeAllDeliveredNotifications();
      if (notificationOpen) {
        const notification = notificationOpen.notification;
        if (notification.data && notification.data.channel && notification.data.channel.channel_url) {
          sb.GroupChannel.getChannel(notification.data.channel.channel_url, (channel, err) => {
            if (!err) {
              this.openChat(channel);
            }
          });
        }
      }
    });
    firebase.notifications().removeAllDeliveredNotifications();
    const connectionHandler = new sb.ConnectionHandler();
    connectionHandler.onReconnectFailed = () => {
      sb.reconnect();
    };
    connectionHandler.onReconnectStarted = () => {};
    connectionHandler.onReconnectSucceeded = () => {
      const manager = SendBirdSyncManager.getInstance();
      manager.resumeSync();
    };
    sb.addConnectionHandler('connection handler', connectionHandler);

    const options = new SendBirdSyncManager.Options();
    options.messageCollectionCapacity = 2000;
    options.messageResendPolicy = 'automatic';
    options.failedMessageRetentionDays = 7;
    options.maxFailedMessageCountPerChannel = 50;
    options.automaticMessageResendRetryCount = 4;
    SendBirdSyncManager.setup(userId, options, err => {
      if (!err) {
        firebase
          .notifications()
          .getInitialNotification()
          .then(notificationOpen => {
            if (notificationOpen) {
              const notification = notificationOpen.notification;
              if (notification.data && notification.data.channel && notification.data.channel.channel_url) {
                sb.GroupChannel.getChannel(notification.data.channel.channel_url, (channel, err) => {
                  if (!err) {
                    this.openChat(channel);
                  }
                });
              }
            }
          });

        const query = sb.GroupChannel.createMyGroupChannelListQuery();
        query.limit = 50;
        query.includeEmpty = false;
        query.order = 'latest_last_message';

        this.collection = new SendBirdSyncManager.ChannelCollection(query);
        const collectionHandler = new SendBirdSyncManager.ChannelCollection.CollectionHandler();
        collectionHandler.onChannelEvent = (action, channels) => {
          switch (action) {
            case 'insert': {
              this.props.addChannels(channels);
              break;
            }
            case 'update': {
              this.props.updateChannels(channels);
              break;
            }
            case 'move': {
              this.props.moveChannels(channels);
              break;
            }
            case 'remove': {
              this.props.removeChannels(channels);
              break;
            }
            case 'clear': {
              this.props.clearChannels();
              break;
            }
          }
        };
        this.collection.setCollectionHandler(collectionHandler);
        this.next();
      } else {
        console.log(err);
      }
    });
  }
  addPushTokenRefreshListener() {
    this.onTokenRefreshListener = firebase.messaging().onTokenRefresh(fcmToken => {
      this.registerPushToken(fcmToken);
    });
  }
  addPushNotificationListener() {
    this.onPushNotificationListener = firebase.messaging().onMessage(message => {
      this.handlePushNotification(message);
    });
  }
  handlePushNotification(message) {
    // DO NOTHING FOR FOREGROUND PUSH NOTIFICATION
  }
  registerPushToken(fcmToken) {
    const sb = SendBird.getInstance();
    if (Platform.OS === 'ios') {
      // WARNING! FCM token doesn't work in request to APNs.
      // Use APNs token here instead.
      firebase
        .messaging()
        .ios.getAPNSToken()
        .then(token => {
          sb.registerAPNSPushTokenForCurrentUser(token, (result, err) => {
            if (err) {
              console.log(err);
            }
          });
        })
        .catch(error => {
          console.error(error);
        });
    } else {
      sb.registerGCMPushTokenForCurrentUser(fcmToken, (result, err) => {
        if (err) {
          console.log(err);
        }
      });
    }
  }
  requireLogin() {
    this.props.navigation.replace('Signin', {});
  }
  createChannel() {
    this.props.navigation.push('Invite', {});
  }
  openChat(channel) {
    if (navigator.top() !== channel.url) {
      this.props.navigation.push('Chat', { channel });
    }
  }
  next() {
    if (this.collection) {
      this.collection.fetch(err => {
        if (err) {
          console.log(err);
        }
      });
    }
  }
  render() {
    return (
      <View style={style.container}>
        <FlatList
          data={this.state.channels}
          keyExtractor={channel => channel.url}
          onEndReached={this.next.bind(this)}
          renderItem={bundle => {
            const channel = bundle.item;
            return <GroupChannelView channel={channel} onPress={this.openChat.bind(this)}></GroupChannelView>;
          }}
        />
        <ActionButton
          buttonColor="#6e5baa"
          degrees={0}
          shadowStyle={style.actionButtonShadow}
          fixNativeFeedbackRadius={true}
          onPress={this.createChannel.bind(this)}
        >
          <Icon name="md-create" style={style.actionButtonIcon} />
        </ActionButton>
      </View>
    );
  }
}

export default connect(
  previousState => {
    const { main } = previousState;
    return { ...main };
  },
  Action
)(MainController);
