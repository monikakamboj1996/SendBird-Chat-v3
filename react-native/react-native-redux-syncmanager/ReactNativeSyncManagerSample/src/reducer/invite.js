import Action from "../action/invite";

const _initialState = {
	users: [],
	err: null
};

export const reducer = (state = _initialState, action) => {
	const newState = { ...state };
	switch(action.type) {
		case Action.Type.USER_FETCH: {
			newState.users = newState.users.concat(action.payload.users);
			break;
		}
	}
	return newState;
};