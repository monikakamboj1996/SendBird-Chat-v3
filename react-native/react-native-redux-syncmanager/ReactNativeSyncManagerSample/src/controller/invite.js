import React from 'react';
import { View } from 'react-native';
import { Button } from 'react-native-elements';
import { connect } from 'react-redux';
import Icon from 'react-native-vector-icons/Ionicons';
import { FlatList } from 'react-native-gesture-handler';

import SendBird from 'sendbird';

import { style } from '../style/invite';
import Action from '../action/invite';
import UserView from '../view/user';
import { toast } from '../utils';

let _selectedUserIds = [];

export class InviteController extends React.Component {
  static navigationOptions = ({ navigation }) => {
    const channel = navigation.getParam('channel', null);
    return {
      title: !channel ? 'Create new channel' : 'Invite',
      headerRight: (
        <View style={{ flexDirection: 'row' }}>
          <Button
            containerStyle={style.headerInviteContainer}
            buttonStyle={style.headerInvite}
            icon={<Icon name="md-add" color="#7d62d9" size={28}></Icon>}
            onPress={() => {
              if (!channel) {
                if (_selectedUserIds.length > 0) {
                  const sb = SendBird.getInstance();
                  const params = new sb.GroupChannelParams();
                  params.addUserIds(_selectedUserIds);
                  sb.GroupChannel.createChannel(params, (channel, err) => {
                    if (!err) {
                      navigation.replace('Chat', { channel });
                    } else {
                      toast('Failed to create channel.');
                    }
                  });
                } else {
                  toast('Choose at least 1 user to invite.');
                }
              } else {
                channel.inviteWithUserIds(_selectedUserIds, (res, err) => {
                  if (!err) {
                    navigation.pop();
                  } else {
                    toast('Failed to invite users.');
                  }
                  isLoading = false;
                });
              }
            }}
          />
        </View>
      )
    };
  };
  constructor(props) {
    super(props);
    const sb = SendBird.getInstance();
    this.query = sb.createUserListQuery();
    this.channel = this.props.navigation.getParam('channel', null);
    this.state = {
      users: []
    };
  }
  static getDerivedStateFromProps(props, state) {
    state.users = props.users;
    return state;
  }
  componentDidMount() {
    this.next();
  }
  componentWillUnmount() {
    _selectedUserIds = [];
  }
  toggle(selected, user) {
    if (selected) {
      if (_selectedUserIds.indexOf(user.userId) < 0) {
        _selectedUserIds.push(user.userId);
      }
    } else {
      const index = _selectedUserIds.indexOf(user.userId);
      if (index >= 0) {
        _selectedUserIds.splice(index, 1);
      }
    }
  }
  next() {
    if (this.query.hasNext) {
      this.query.next((users, err) => {
        if (!err) {
          let filteredUsers = [];
          if (this.channel) {
            for (let i in users) {
              if (this.channel.members.map(m => m.userId).indexOf(users[i].userId) < 0) {
                filteredUsers.push(users[i]);
              }
            }
          } else {
            filteredUsers = users;
          }
          if (filteredUsers.length < this.query.limit && this.query.hasNext) {
            setTimeout(() => this.next(), 200);
          } else {
            this.props.fetchUsers(filteredUsers);
          }
        }
      });
    }
  }
  render() {
    return (
      <View>
        <FlatList
          data={this.state.users}
          keyExtractor={user => user.userId + ''}
          onEndReached={this.next.bind(this)}
          renderItem={bundle => (
            <UserView
              user={bundle.item}
              selectable={true}
              selected={_selectedUserIds.indexOf(bundle.item.userId) >= 0}
              onPress={this.toggle.bind(this)}
            />
          )}
        />
      </View>
    );
  }
}

export default connect(
  previousState => {
    const { invite } = previousState;
    return { ...invite };
  },
  Action
)(InviteController);
