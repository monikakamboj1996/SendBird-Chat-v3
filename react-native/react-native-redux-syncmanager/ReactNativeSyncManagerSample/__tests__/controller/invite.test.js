import 'react-native';
import React from 'react';
import { InviteController } from '../../src/controller/invite';
import renderer from 'react-test-renderer';

const members = [
  {
    userId: 1,
    nickname: 'Aaa'
  },
  {
    userId: 2,
    nickname: 'Baa'
  },
  {
    userId: 3,
    nickname: 'Caa'
  }
];

const channel = {
  members: [
    {
      userId: 1
    },
    {
      userId: 4
    }
  ]
};

const navigation = {
  setParams: jest.fn(),
  getParam: jest.fn().mockReturnValue(channel),
  navigate: jest.fn(),
  state: {
    params: {
      channelUrl: 'channelUrl',
      isOpenChannel: true
    }
  },
  dispatch: jest.fn(),
  addListener: jest.fn()
};

describe('controller/invite', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<InviteController users={members} navigation={navigation} fetchUsers={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO: add test cases
});
