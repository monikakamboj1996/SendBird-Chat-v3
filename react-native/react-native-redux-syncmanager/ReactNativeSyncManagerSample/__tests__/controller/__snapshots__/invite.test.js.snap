// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controller/invite renders correctly 1`] = `
<View>
  <RCTScrollView
    collapsable={false}
    data={
      Array [
        Object {
          "nickname": "Aaa",
          "userId": 1,
        },
        Object {
          "nickname": "Baa",
          "userId": 2,
        },
        Object {
          "nickname": "Caa",
          "userId": 3,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
              }
            }
          >
            <Text
              allowFontScaling={false}
              containerStyle={
                Object {
                  "margin": 0,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "padding": 0,
                }
              }
              iconStyle={
                Object {
                  "margin": 0,
                  "padding": 0,
                }
              }
              style={
                Array [
                  Object {
                    "color": "#e3e3e3",
                    "fontSize": 28,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ïŠ»
            </Text>
            <View
              width={16}
            />
            <View
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 17,
                  "height": 34,
                  "width": 34,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#bdbdbd",
                    "borderRadius": 17,
                    "flex": 1,
                    "overflow": "hidden",
                    "position": "relative",
                  }
                }
              >
                <Image
                  onLoadEnd={[Function]}
                  source={
                    Object {
                      "testUri": "../../../src/img/icon_sb_68.png",
                    }
                  }
                  style={
                    Object {
                      "flex": 1,
                      "height": null,
                      "width": null,
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#bdbdbd",
                        "flex": 1,
                        "height": null,
                        "justifyContent": "center",
                        "width": null,
                      }
                    }
                    testID="RNE__Image__placeholder"
                  />
                </View>
              </View>
            </View>
            <View
              width={16}
            />
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginLeft": 8,
                  }
                }
                testID="listItemTitle"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                Aaa
              </Text>
            </View>
            <View
              width={16}
            />
            <Text />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
              }
            }
          >
            <Text
              allowFontScaling={false}
              containerStyle={
                Object {
                  "margin": 0,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "padding": 0,
                }
              }
              iconStyle={
                Object {
                  "margin": 0,
                  "padding": 0,
                }
              }
              style={
                Array [
                  Object {
                    "color": "#e3e3e3",
                    "fontSize": 28,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ïŠ»
            </Text>
            <View
              width={16}
            />
            <View
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 17,
                  "height": 34,
                  "width": 34,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#bdbdbd",
                    "borderRadius": 17,
                    "flex": 1,
                    "overflow": "hidden",
                    "position": "relative",
                  }
                }
              >
                <Image
                  onLoadEnd={[Function]}
                  source={
                    Object {
                      "testUri": "../../../src/img/icon_sb_68.png",
                    }
                  }
                  style={
                    Object {
                      "flex": 1,
                      "height": null,
                      "width": null,
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#bdbdbd",
                        "flex": 1,
                        "height": null,
                        "justifyContent": "center",
                        "width": null,
                      }
                    }
                    testID="RNE__Image__placeholder"
                  />
                </View>
              </View>
            </View>
            <View
              width={16}
            />
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginLeft": 8,
                  }
                }
                testID="listItemTitle"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                Baa
              </Text>
            </View>
            <View
              width={16}
            />
            <Text />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
              }
            }
          >
            <Text
              allowFontScaling={false}
              containerStyle={
                Object {
                  "margin": 0,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "padding": 0,
                }
              }
              iconStyle={
                Object {
                  "margin": 0,
                  "padding": 0,
                }
              }
              style={
                Array [
                  Object {
                    "color": "#e3e3e3",
                    "fontSize": 28,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ïŠ»
            </Text>
            <View
              width={16}
            />
            <View
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 17,
                  "height": 34,
                  "width": 34,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#bdbdbd",
                    "borderRadius": 17,
                    "flex": 1,
                    "overflow": "hidden",
                    "position": "relative",
                  }
                }
              >
                <Image
                  onLoadEnd={[Function]}
                  source={
                    Object {
                      "testUri": "../../../src/img/icon_sb_68.png",
                    }
                  }
                  style={
                    Object {
                      "flex": 1,
                      "height": null,
                      "width": null,
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#bdbdbd",
                        "flex": 1,
                        "height": null,
                        "justifyContent": "center",
                        "width": null,
                      }
                    }
                    testID="RNE__Image__placeholder"
                  />
                </View>
              </View>
            </View>
            <View
              width={16}
            />
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "marginLeft": 8,
                  }
                }
                testID="listItemTitle"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                Caa
              </Text>
            </View>
            <View
              width={16}
            />
            <Text />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
